src/main/webapp/app/admin/statistics/statistics.component.ts

```diff
@@ -0,0 +1,178 @@
+import { Component, OnInit, OnChanges, ViewChild } from '@angular/core';
+import { StatisticsService } from 'app/admin/statistics/statistics.service';
+import { SPAN_PATTERN } from 'app/app.constants';
+import { ChartDataSets, ChartOptions, ChartType } from 'chart.js';
+import { BaseChartDirective, Label } from 'ng2-charts';
+import { DataSet } from 'app/exercises/quiz/manage/statistics/quiz-statistic/quiz-statistic.component';
+import { TranslateService } from '@ngx-translate/core';
+import * as moment from 'moment';
+
+export enum SpanType {
+    DAY = 'DAY',
+    WEEK = 'WEEK',
+    MONTH = 'MONTH',
+    YEAR = 'YEAR',
+}
+
+@Component({
+    selector: 'jhi-statistics',
+    templateUrl: './statistics.component.html',
+})
+export class JhiStatisticsComponent implements OnInit, OnChanges {
+    spanPattern = SPAN_PATTERN;
+    span: SpanType = SpanType.WEEK;
+
+    // Histogram related properties
+    public histogramData: number[] = [];
+    public barChartOptions: ChartOptions = {};
+    public barChartLabels: Label[] = [];
+    public barChartType: ChartType = 'bar';
+    public barChartLegend = true;
+    public UserLoginChartData: ChartDataSets[] = [];
+    public SubmissionsChartData: ChartDataSets[] = [];
+
+    @ViewChild(BaseChartDirective) chart: BaseChartDirective;
+
+    constructor(private service: StatisticsService, private translateService: TranslateService) {}
+
+    async ngOnInit() {
+        await this.setBinWidth();
+        await this.createChart();
+    }
+
+    async ngOnChanges() {}
+
+    private async setBinWidth(): Promise<void> {
+        switch (this.span) {
+            case SpanType.DAY:
+                this.histogramData = Array(24).fill(0);
+                break;
+            case SpanType.WEEK:
+                this.histogramData = Array(7).fill(0);
+                break;
+            case SpanType.MONTH:
+                const days = this.daysInMonth();
+                this.histogramData = Array(days).fill(0);
+                break;
+            case SpanType.YEAR:
+                this.histogramData = Array(12).fill(0);
+                break;
+        }
+    }
+
+    private daysInMonth(): number {
+        return new Date(new Date().getFullYear(), new Date().getMonth(), 0).getDate();
+    }
+
+    async getSubmissions(): Promise<number[]> {
+        return new Promise<number[]>((resolve, reject) => {
```

Comment:
Why are you using a Promise here? Can't we work with Observables here as well?

Reply:
The graph was initialized before the Submissions got fetched, so I used asynchronous method calls to wait for the result

Reply:
But why not do it using observables? @FuchsDominik . It is highly unusual in angular to work with promises as everything is based on observables

Reply:
Well i still work with observables. All i can say is that using an explicit Promise did the trick as through asynchronous method calls the graph is not instantiated before the result comes back.

Reply:
@FuchsDominik Lets have a look together. Maybe some RxJs magic can help :)

Reply:
Resolved in [this](https://github.com/ls1intum/Artemis/pull/2471/commits/f6e30d7546138808db3176d23669807099621dc0) commit